userchroot.c

```diff
@@ -2,15 +2,13 @@
 #include <linux/version.h>
 #endif
 
-#if defined (__linux__) && defined (MOUNT_PROC)
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2,6,24))
```

Comment:
Is this being removed because it was redundant, or because we are not going to support such old versions of the kernel?

Reply:
Mainly the latter (2.6.24 is almost 12 years old), however, such Linux kernel version checks in userspace are not ideal anyway.

For one, they only check the version of the kernel header files. The header files may or may not have the same version as the running kernel and there are also many cases where the version of the running kernel of the build machine doesn't match the version of the target kernel (e.g., CI).

Also, especially for security sensitive setuid root code, I think it's problematic to implicitly disable an isolation feature just because the kernel headers on the build machine are too old.

If we needed to keep supporting kernels older than 2.6.24, it would be better to fail the build when `MOUNT_PROC` is defined instead of silently disabling it, in my opinion.

Does userchroot need to run on kernels older than 2.6.24 (RHEL 5 is the only possibility that comes to mind)?

Reply:
Right, this was added here when we were still supporting RHEL5. I don't think we need to care about RHEL5 anymore, so I think making this non-optional is ok.

Reply:
OK makes sense, thank you both.